cs/src/core/Bond.csproj

```diff
@@ -1,101 +1,36 @@
-    <Compile Include="expressions\SerializerTransform.cs" />
-    <Compile Include="expressions\StringExpression.cs" />
-    <Compile Include="expressions\TaggedParser.cs" />
-    <Compile Include="expressions\TaggedReader.cs" />
-    <Compile Include="expressions\ThrowExpression.cs" />
-    <Compile Include="expressions\Transform.cs" />
-    <Compile Include="expressions\TypeAlias.cs" />
-    <Compile Include="expressions\UntaggedParser.cs" />
-    <Compile Include="expressions\UntaggedReader.cs" />
-    <Compile Include="expressions\xml\SimpleXmlParser.cs" />
-    <Compile Include="expressions\xml\XmlParser.cs" />
-    <Compile Include="expressions\xml\XmlReader.cs" />
-    <Compile Include="Field.cs" />
-    <Compile Include="GenericFactory.cs" />
-    <Compile Include="IFactory.cs" />
-    <Compile Include="io\ICloneable.cs" />
-    <Compile Include="io\IInputStream.cs" />
-    <Compile Include="io\IntegerHelper.cs" />
-    <Compile Include="io\IOutputStream.cs" />
-    <Compile Include="io\safe\InputBuffer.cs" />
-    <Compile Include="io\safe\OutputBuffer.cs" />
-    <Compile Include="ISchemaField.cs" />
-    <Compile Include="Marshaler.cs" />
-    <Compile Include="properties\AssemblyInfo.cs" />
-    <Compile Include="Property.cs" />
-    <Compile Include="protocols\CompactBinary.cs" />
-    <Compile Include="protocols\CompactBinaryCounter.cs" />
-    <Compile Include="protocols\Exceptions.cs" />
-    <Compile Include="protocols\FastBinary.cs" />
-    <Compile Include="protocols\IProtocolWriter.cs" />
-    <Compile Include="protocols\ITaggedReader.cs" />
-    <Compile Include="protocols\IUntaggedReader.cs" />
-    <Compile Include="protocols\IXmlReader.cs" />
-    <Compile Include="protocols\SimpleBinary.cs" />
-    <Compile Include="protocols\SimpleXmlReader.cs" />
-    <Compile Include="protocols\SimpleXmlWriter.cs" />
-    <Compile Include="protocols\XmlMetadata.cs" />
-    <Compile Include="Reflection.cs" />
-    <Compile Include="RuntimeSchema.cs" />
-    <Compile Include="Schema.cs" />
-    <Compile Include="Serializer.cs" />
-    <Compile Include="Transcoder.cs" />
-    <!-- Resharper Workaround -->
-    <Compile Include="$(IntermediateOutputPath)\bond_types.cs" Condition="False" />
-    <Compile Include="$(IntermediateOutputPath)\bond_const_types.cs" Condition="False" />
-    <!-- End Resharper Workaround -->
+    <Compile Condition="False" Include="$(IntermediateOutputPath)\bond_types.cs" />
+    <Compile Condition="False" Include="$(IntermediateOutputPath)\bond_const_types.cs" />
   </ItemGroup>
+
   <ItemGroup>
-    <Reference Include="Bond.Attributes">
-      <HintPath>..\attributes\$(DependentOutputPath)\Bond.Attributes.dll</HintPath>
-    </Reference>
-    <Reference Include="Bond.Reflection">
-      <HintPath>..\reflection\$(DependentOutputPath)\Bond.Reflection.dll</HintPath>
-    </Reference>
+    <ProjectReference Include="..\attributes\Attributes.csproj" />
```

Comment:
Consider defining a property in Common.Internal.props that points to the "src" directory and writing these project references to the core Bond libraries in terms of that. It will reduce the number of "../.." that we need and make moving projects around easier.

Something like this in Common.Internal.props:

```xml
<PropertyGroup>
    <BondRoot>$([MSBuild]::NormalizeDirectory("$(MSBuildThisFileDirectory)..\..\"))</BondRoot>
</PropertyGroup>
```

Then the project references everywhere become:

```xml
<ProjectReference Include="$(BondRoot)attributes\Attributes.csproj" />
```

This helps more in the test projects.